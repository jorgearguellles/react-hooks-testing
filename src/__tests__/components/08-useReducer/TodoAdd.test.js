import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { demoTodos } from '../../fixtures/demoTodos';
import { TodoAdd } from '../../../components/08-useReducer/TodoAdd';

describe('Testing <TodoAdd />', ()=>{

  const handleAddTodo = jest.fn();
  const wrapper = shallow(
    <TodoAdd 
      handleAddTodo={ handleAddTodo }
    />
  );

  test('1. SnapShot', ()=>{
    expect(wrapper).toMatchSnapshot();
  })

  test("2. Shouldn't call: handleAddTodo ", ()=>{

    const formSubmit = wrapper.find('form').prop('onSubmit');
    formSubmit( { preventDefault(){} });
    expect( handleAddTodo ).toHaveBeenCalledTimes(0);
    expect( handleAddTodo ).not.toHaveBeenCalled();
  })

  test('3. Should call: handleAddTodo ', ()=>{
    const value = 'Learn Node.js';
    wrapper.find('input').simulate('change', {
      target: {
        value,
        name: 'description',
      }
    })
    
    const formSubmit = wrapper.find('form').prop('onSubmit');
    formSubmit( { preventDefault(){} });
    expect( handleAddTodo ).toHaveBeenCalled();
    expect( handleAddTodo ).toHaveBeenCalledTimes(1);
    expect( handleAddTodo ).toHaveBeenCalledWith( expect.any(Object) ) // Any Object { } pass test
    expect( handleAddTodo ).toHaveBeenCalledWith( {
      id: expect.any(Number), // Any number pass test
      desc: value,
      done: false,
    } )
    expect( wrapper.find('input').prop('value') ).toBe('');
  });

})