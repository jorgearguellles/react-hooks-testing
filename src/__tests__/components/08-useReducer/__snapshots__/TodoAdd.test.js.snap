// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <TodoAdd /> 1. SnapShot 1`] = `
<Fragment>
  <h4>
    Add ToDo
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="Learn ..."
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-primary mt-1 btn-block"
      type="submit"
    >
      Add
    </button>
  </form>
</Fragment>
`;
